src/PortingAssistant.Client.Telemetry/Uploader.cs

```diff
@@ -12,130 +10,251 @@
+            _client = telemetryClient;
+            ReadFileLineMap();
+            GetLogName = GetLogNameDefault;
+        }
+
+        /// <summary>
+        /// List of errors during upload actions. Key is error category and Value is Exception
+        /// </summary>
+        public readonly Dictionary<string, Exception> ErrorList = new();
+        public Func<string, string> GetLogName { get; set; }
+
+        private static string GetLogNameDefault(string file)
+        {
+            var logName = Path.GetFileNameWithoutExtension(file);
+            var fileExtension = Path.GetExtension(file);
+            string logNameWithoutDate = int.TryParse(logName.Split('-').LastOrDefault() ?? "", out _) ? string.Join('-', logName.Split('-').SkipLast(1)) : logName;
+            if (fileExtension == ".log")
+            {
+                logName = $"{logNameWithoutDate}-logs";
+            }
+            else if (fileExtension == ".metrics")
+            {
+                logName = $"{logNameWithoutDate}-metrics";
+            }
+            return logName;
+        }
+
+        public bool Upload(IEnumerable<string> fileEntries)
         {
             try
             {
-                var fileEntries = Directory.GetFiles(teleConfig.LogsPath)
-                    .Where(f =>
-                        Path.GetFileName(f)
-                            .StartsWith("portingAssistant-client-cli") &&
-                        teleConfig.Suffix.ToArray().Any(x => f.EndsWith(x))
-                    ).ToList();
-
-                var lastReadTokenFile = Path.Combine(teleConfig.LogsPath, "lastToken.json");
-                var fileLineNumberMap = new Dictionary<string, int>();
-                if (File.Exists(lastReadTokenFile))
+                foreach (var file in fileEntries)
                 {
-                    fileLineNumberMap = JsonConvert.DeserializeObject<Dictionary<string, int>>(File.ReadAllText(lastReadTokenFile));
+                    var logName = GetLogName(file);
+                    if (string.IsNullOrEmpty(logName))
+                    {
+                        continue;
+                    }
+                    if (_client != null)
+                    {
+                        UploadFile(file, logName);
+                    }
                 }
-                foreach (var file in fileEntries)
+                if (_updatedFileLineNumberMap.Any())
+                {
+                    UpdateFileLineMapJson();
+                }
+                return true;
+            }
+            catch (Exception ex)
+            {
+                ErrorList.Add("UploadFilesError", ex);
+                return false;
+            }
+        }
+
+        public void CleanupLogFolder()
+        {
+            try
+            {
+                long folderSizeLimit = _configuration.LogsFolderSizeLimit != 0
+                    ? _configuration.LogsFolderSizeLimit
+                    : 250000000;
```

Comment:
Move this to a config class so we can change if needed

Reply:
part of config file already, this is just a sanity check to make sure the size limit has default value of 250 MB, will move to private const variable as per our discussion